### PR TITLE
Pbi 134 dropdown

### PR DESCRIPTION
**Pull request by bryannaufal**
## Describe your changes
Create unit test and implement Searchable Dropdown component

## Issue ticket number and link
ADU-134: https://linear.app/adudu/issue/ADU-134/dropdown

### CODE REVIEW

File: src/components/atoms/dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  const [searchTerm, setSearchTerm] = useState<string>('')
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [selectedOption, setSelectedOption] = useState<string | null>(null)
+
+  const filteredOptions = options.filter(option =>
+    option.toLowerCase().includes(searchTerm.toLowerCase()),
+  )
+
+  function handleOptionClick(option: string) {
+    console.log('Selected option:', option)
```
Comment (edutjie) [inline src/components/atoms/dropdown.tsx]: Is this for debugging? If yes, don't forget to remove it!

**Comment by valenciusap17** [comment]
Is this dropdown refering to the component library that we are using, Shadcn UI?

**Comment by bryannaufal** [comment]
> Is this dropdown refering to the component library that we are using, Shadcn UI?

Sorry there was a mistake in the component implementation (not using the Shadcn UI component library), currently being fixed to use Shadcn UI

File: src/components/atoms/dropdown.tsx (hunk 1 above)
Comment (FreeJ1nG) [reply src/components/atoms/dropdown.tsx]: CC: @bryannaufal

**Comment by FreeJ1nG** [comment]
> > Is this dropdown refering to the component library that we are using, Shadcn UI?
> 
> Sorry there was a mistake in the component implementation (not using the Shadcn UI component library), currently being fixed to use Shadcn UI

I see that the component is still self-made, are you going to fix this?

File: src/routes/vendor.tsx [hunk 2]
```diff
@@ -36,6 +37,13 @@ export default function VendorPage() {
     setShowLoadingOverlay(!isSuccess)
   }, [isSuccess, setShowLoadingOverlay])
 
+  // TODO: Change with real location
+  const locationOptions = [
+    { value: 'jakarta', label: 'Jakarta' },
+    { value: 'bandung', label: 'Bandung' },
+    { value: 'surabaya', label: 'Surabaya' },
+  ]
```
Comment (edutjie) [inline src/routes/vendor.tsx:45]: When are you planning to integrate this using the API?

**Comment by edutjie** [comment]
image

Idk if it's already looks good on your end, but in my end it the filter by product input field is not full.

**Comment by FreeJ1nG** [comment]
LGTM, we'll do the integration part in another PR

Merging

**Pull request merged**